sklearn/pipeline.py

```diff
@@ -177,19 +177,25 @@ def _validate_steps(self):
         for t in transformers:
             if t is None or t == 'passthrough':
                 continue
-            if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
-                    hasattr(t, "transform")):
+            if (not hasattr(t, "fit_transform") or
+                    (not hasattr(t, "fit") and hasattr(t, "transform"))):
```

Comment:
It should be 
```python
(not (hasattr(t, "fit") and hasattr(t, "transform")))):
```
no? I think that you need parentheses to say `not (fit and transform)`.

Reply:
Yes, that's correct, not has precedence over and. I'll fix it